src/Mod/TechDraw/App/Geometry.cpp

```diff
@@ -449,7 +447,7 @@ bool BaseGeom::closed(void)
 
 
 //! Convert 1 OCC edge into 1 BaseGeom (static factory method)
-BaseGeom* BaseGeom::baseFactory(TopoDS_Edge edge)
+BaseGeomPtr BaseGeom::baseFactory(TopoDS_Edge edge)
 {
     std::unique_ptr<BaseGeom> result;
```

Comment:
BaseGeomPtr is declared as std::shared_ptr so result should be a shared_ptr too, although a shared_ptr accepts a unique_ptr.